File: app/src/main/java/com/arcao/feedback/collector/SharedPreferencesCollector.java
package com.arcao.feedback.collector;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Map;

public class SharedPreferencesCollector extends Collector {
    private final Context context;
    private final String sharedPreferenceName;

    public SharedPreferencesCollector(Context context) {
        this(context, null);
    }

    public SharedPreferencesCollector(Context context, String sharedPreferenceName) {
        this.context = context.getApplicationContext();
        this.sharedPreferenceName = sharedPreferenceName;
    }

    @Override
    public String getName() {
        if (sharedPreferenceName == null)
            return "SharedPreferences.default";

        return "SharedPreferences." + sharedPreferenceName;
    }

    @Override
    protected String collect() {
        final SharedPreferences prefs;

        if (sharedPreferenceName == null) {
            prefs = PreferenceManager.getDefaultSharedPreferences(context);
        } else {
            prefs = context.getSharedPreferences(sharedPreferenceName, Context.MODE_PRIVATE);
        }

        final Map<String, ?> prefEntries = prefs.getAll();

        // In case it's empty return empty text
        if (prefEntries.isEmpty()) {
            return "Empty";
        }

        StringBuilder result = new StringBuilder();
        // Add all preferences
        for (final String key : prefEntries.keySet()) {
            final Object prefValue = prefEntries.get(key);
            result.append(key).append('=');
            result.append(prefValue == null ? "null" : prefValue.toString());
            result.append("\n");
        }
        return result.toString();
    }
}

File: app/src/main/java/com/arcao/feedback/collector/SharedPreferencesCollector.kt
package com.arcao.feedback.collector

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SharedPreferencesCollector @JvmOverloads constructor(private val context: Context, private val sharedPreferenceName: String? = null) : Collector() {
    override val name: String
        get() = if (sharedPreferenceName == null) "SharedPreferences.default" else "SharedPreferences.$sharedPreferenceName"

    override suspend fun collect(): String {
        val prefs: SharedPreferences = if (sharedPreferenceName == null) {
            PreferenceManager.getDefaultSharedPreferences(context)
        } else {
            context.getSharedPreferences(sharedPreferenceName, Context.MODE_PRIVATE)
        }

        val entries = prefs.all

        // In case it's empty return empty text
        if (entries.isEmpty()) {
            return "Empty"
        }

        val result = StringBuilder()
        // Add all preferences
        entries.forEach { (key, value) ->
            result.append(key)
                    .append('=')
                    .append(value.toString())
                    .append("\n")
        }
        return result.toString()
    }
}
